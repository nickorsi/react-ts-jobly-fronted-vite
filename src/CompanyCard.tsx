import React from "react";
import { CompanyDataFromAPI } from "./CompanyList";

/**
 * CompanyCard component that renders information about the company.
 *
 * Takes props {handle, name, description, numEmployees, logoURL} as props
 *
 * CompanyList -> CompanyCard
 */
function CompanyCard(
    {handle, name, description, logoUrl}: CompanyDataFromAPI
) {
    console.debug(
        'CompanyCard',
        'handle=',
        handle,
        'name=',
        name,
        'description=',
        description,
        'logoURL',
        logoUrl
    )
    return (
        <a href={`companies/${handle}`}>
            <div className="company-card">
                <div className="company-brand">
                    <p>{ name }</p>
                    {
                        logoUrl === 'null' ?
                        null :
                        <img src={`${logoUrl}`} alt="company logo" />
                    }
                </div>
                <div className="company-info">
                    <p>{ description }</p>
                </div>
            </div>
        </a>
    )
}

export default CompanyCard;