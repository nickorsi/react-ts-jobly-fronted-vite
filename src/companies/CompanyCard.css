.company {
  text-decoration: none;
}

.company:hover {
  box-shadow: 0px 0px 5px 5px rgb(10, 173, 214);
  text-decoration: underline;
}

.company:active {
  box-shadow: 0px 0px 0px 0px rgb(10, 173, 214);
  text-decoration: none;
}

.company-info p {
  font-size: 1.1rem;
  margin-top: 10px;
}

.card {
  background-color: white;
  border-radius: 10px;
  padding: 15px;
  margin: 15px;
}

.company-brand {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

@media screen and (max-width: 700px) {
  .company-brand {
    flex-direction: column;
  }
}

.company-brand img {
  width: 100px;
  border: solid black 2px;
  border-radius: 5px;
}