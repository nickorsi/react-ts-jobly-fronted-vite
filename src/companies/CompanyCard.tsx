import React from "react";
import { CompanyDataFromAPI } from "./CompanyList";
import './CompanyCard.css';
import { Link } from 'react-router-dom'

/**
 * CompanyCard component that renders information about the company.
 *
 * Takes props {handle, name, description, numEmployees, logoURL} as props
 *
 * CompanyList -> CompanyCard
 */
function CompanyCard(
    {handle, name, description, numEmployees, logoUrl}: CompanyDataFromAPI
) {

    return (
        <Link className="company card" to={`${handle}`} data-testid="company-link">
            <div className="company-info">
                <div className="company-brand">
                    <p>{ name }</p>
                    {
                        logoUrl === 'null' ? //Comparing to null string, passed as prop
                        null :
                        <img src={`${logoUrl}`} alt="company logo" />
                    }
                </div>
                <p>{ description }</p>
            </div>
        </Link>
    )
}

export default CompanyCard;