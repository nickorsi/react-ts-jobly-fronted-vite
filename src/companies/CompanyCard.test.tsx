import {describe, expect, it} from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import CompanyCard from "./CompanyCard.tsx";

const defaultCompany = {
    "handle": "c1",
    "name": "Company 1",
    "description": "Description 1",
    "numEmployees": 1,
    "logoUrl": "http//c1.img",
}


describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(
            <MemoryRouter>
                <CompanyCard
                    handle={defaultCompany.handle}
                    name={defaultCompany.name}
                    description={defaultCompany.description}
                    numEmployees={defaultCompany.numEmployees}
                    logoUrl={defaultCompany.logoUrl}
                />
            </MemoryRouter>
        );
    });
    it("matches the snapshot", function () {
        const { container } = render(
            <MemoryRouter>
                <CompanyCard
                    handle={defaultCompany.handle}
                    name={defaultCompany.name}
                    description={defaultCompany.description}
                    numEmployees={defaultCompany.numEmployees}
                    logoUrl={defaultCompany.logoUrl}
                />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});