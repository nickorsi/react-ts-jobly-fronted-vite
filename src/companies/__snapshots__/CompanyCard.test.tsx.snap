// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <a
    class="company card"
    href="/c1"
  >
    <div
      class="company-info"
    >
      <div
        class="company-brand"
      >
        <p>
          Company 1
        </p>
        <img
          alt="company logo"
          src="http//c1.img"
        />
      </div>
      <p>
        Description 1
      </p>
    </div>
  </a>
</div>
`;
